libminifi/include/core/PropertyType.h

```diff
@@ -90,12 +47,19 @@ class PropertyValidator {
   [[nodiscard]] virtual ValidationResult validate(const std::string &subject, const std::shared_ptr<minifi::state::response::Value> &input) const = 0;
 
   [[nodiscard]] virtual ValidationResult validate(const std::string &subject, const std::string &input) const = 0;
+};
+
+class PropertyType : public PropertyParser, public PropertyValidator {
```

Comment:
I think composition would be better design here, than multiple inheritance.

Most property types have names changed to end with "_TYPE", except `UnsignedIntPropertyType`. This particular case may be an oversight.

But the pattern ultimately overrides the `getName` method of `PropertyValidator`, so it should retuirn a validator name (like the old version), but a method called `getName` in `PropertyType` has no business returning a validator name. I'm also not sure how the type name gets converted back to a validator name when used with a validator-aware C2 server.

Reply:
On the 2nd part: good catch, thanks!  We can't change the validator names we send to C2.  I have reverted them in 1f9cd0bf0dd802b61175e670dba59d4e0ca2fb25.

On the 1st part: this is not really multiple inheritance, just a separation of two interfaces.  Are you sure we want to replace 12 classes, each implementing two interfaces, with 36 classes using composition?

Reply:
as discussed offline, I have renamed `getName()` to `getValidatorName()` in b052e55ef42515d9ef441975cf1188e95c9625f0